packages/amplify-migration-codegen-e2e/src/assertions.ts

```diff
@@ -5,50 +5,108 @@ import {
+}
 
+async function assertStorage(gen1Meta: $TSAny, gen1Region: string) {
+  const { BucketName: gen1BucketName } = Object.keys(gen1Meta.storage).map((key) => gen1Meta.storage[key])[0].output;
   expect(gen1BucketName).toBeDefined();
   const bucketExists = await checkIfBucketExists(gen1BucketName, gen1Region);
   expect(bucketExists).toMatchObject({});
+  return { gen1BucketName };
+}
+
+async function assertAPI(gen1Meta: $TSAny, gen1Region: string) {
+  const {
+    GraphQLAPIIdOutput: gen1GraphqlApiId,
+    GraphQLAPIEndpointOutput,
+    GraphQLAPIKeyOutput,
+  } = Object.keys(gen1Meta.api).map((key) => gen1Meta.api[key])[0].output;
+  const { graphqlApi } = await getAppSyncApi(gen1GraphqlApiId, gen1Region);
 
   expect(gen1GraphqlApiId).toBeDefined();
   expect(GraphQLAPIEndpointOutput).toBeDefined();
   expect(GraphQLAPIKeyOutput).toBeDefined();
 
   expect(graphqlApi).toBeDefined();
   expect(graphqlApi?.apiId).toEqual(gen1GraphqlApiId);
+  return { gen1GraphqlApiId };
+}
+
+export async function assertDefaultGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  const { gen1GraphqlApiId } = await assertAPI(gen1Meta, gen1Region);
+
   return { gen1UserPoolId, gen1FunctionName, gen1BucketName, gen1GraphqlApiId, gen1Region };
 }
 
+export async function assertAuthWithMaxOptionsGen1Setup(projRoot: string) {
+  const { gen1Meta, gen1Region } = await getMeta(projRoot);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+
+  const {
+    UserPoolId: userPoolId,
+    AppClientIDWeb: appClientIdWeb,
+    AppClientID: appClientId,
+  } = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output;
+  const clientIds = [appClientIdWeb, appClientId];
+  const clients = await getUserPoolClients(userPoolId, clientIds, gen1Region);
+  expect(clients).toHaveLength(2);
+
+  const { userPoolGroups } = gen1Meta.auth;
+  expect(userPoolGroups.service).toEqual('Cognito-UserPool-Groups');
+  expect(userPoolGroups.providerPlugin).toEqual('awscloudformation');
+  expect(userPoolGroups.dependsOn.length).toBe(1);
+  expect(userPoolGroups.dependsOn[0].category).toBe('auth');
+  expect(userPoolGroups.dependsOn[0].attributes.length).toBe(4);
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('UserPoolId');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientIDWeb');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientID');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('IdentityPoolId');
+
+  return { gen1UserPoolId, gen1FunctionName, gen1Region };
+}
+
+export async function assertStorageWithMaxOptionsGen1Setup(projRoot: string) {
```

Comment:
I don't think you need this wrapper, just call getMeta and assertStorage directly